cluster-autoscaler/core/scale_down.go

```diff
@@ -1101,11 +1103,38 @@ func (sd *ScaleDown) scheduleDeleteEmptyNodes(emptyNodes []*apiv1.Node, client k
 				metrics.RegisterScaleDown(1, gpu.GetGpuTypeForMetrics(sd.context.CloudProvider.GPULabel(), sd.context.CloudProvider.GetAvailableGPUTypes(), nodeToDelete, nodeGroupForDeletedNode), metrics.Unready)
 			}
 			result = status.NodeDeleteResult{ResultType: status.NodeDeleteOk}
-		}(node, nodeGroup)
+		}(node, nodeGroup, sd.context.DaemonSetEvictionForEmptyNodes)
 	}
 	return deletedNodes, nil
 }
 
+// create eviction object for all DaemonSet pods on the node
+func daemonSetEviction(clusterSnapshot simulator.ClusterSnapshot, nodeToDelete *apiv1.Node, client kube_client.Interface, dsEvictionTimeout time.Duration,
+	recorder kube_record.EventRecorder, maxGracefulTerminationSec int) {
+	nodeInfo, _ := clusterSnapshot.NodeInfos().Get(nodeToDelete.Name)
```

Comment:
I don't think we should ignore the returned error here or in the next line. I'd return an error from this function, check for it in `scheduleDeleteEmptyNodes` and just log a warning/error if anything's wrong.